subprocess/path.go

```diff
@@ -52,6 +53,10 @@ func LookPath(file string) (string, error) {
 	path := os.Getenv("PATH")
 	for _, dir := range filepath.SplitList(path) {
 		if dir == "" {
+			// Avoid using current directory on Windows
+			if runtime.GOOS == "windows" && file == "git" {
```

Comment:
I don't think we want to restrict this to just `git`; we'd instead want to make it more generally applicable, so different binaries don't have different behavior.

And I think we'd also want some tests for this behavior, probably in `t/t-path.sh`.